languages/csharp/exercises/concept/regular-expressions/.meta/Example.cs

```diff
@@ -0,0 +1,80 @@
+using System.Text.RegularExpressions;
+
+public class LogParser
+{
+    public bool IsMatch(string text)
+    {
+        const string searchArg = @"\[TRC\] | \[DBG\] | \[INF\] | \[ERR\] | \[WRN\] | \[FTL\]";
+        return Regex.Match(text, searchArg, RegexOptions.IgnorePatternWhitespace).Success;
+    }
+
+    public string[] SplitLogLine(string text)
+    {
+        return Regex.Split(text, "<[*^=-]*>");
+    }
+
+    public bool[] AreQuotedPasswords(string[] lines)
+    {
+        bool[] results = new bool[lines.Length];
+        var regex = new Regex(@"^.*""[^\\""]*password[^\\""]*"".*$", RegexOptions.IgnoreCase);
+        int ctr = 0;
+        for (int i = 0; i < lines.Length; i++)
+        {
+            results[i] = regex.IsMatch(lines[i]);
+        }
+
+        return results;
+    }
+
+    public string RemoveEndOfLineText(string line)
+    {
+        string pattern = @"end-of-line\d+";
+
+        string str = Regex.Replace(line, pattern, string.Empty,
+            RegexOptions.IgnoreCase | RegexOptions.IgnorePatternWhitespace);
+        return str;
+    }
+
+    public string[] RewriteLogLines(string[] lines)
```

Comment:
I feel that this implementation might be a tad on the complex side. I expect there to be a very large number of representations due to its complexity. Maybe we could simplify things a bit? The most important thing this method teaches the student is about capture groups, so focusing on just that might be nice.

Reply:
> I think that will be a challenge with regex.

Yeah, I agree. This will be problematic :)